File: Data Structures & Algorithms/maximum-subarray/submission-3.java
class Solution {
    public int maxSubArray(int[] nums) {
        int curSum = 0;
        int maxSum = nums[0];

        for (int num: nums) {
            curSum = Math.max(curSum + num, num);
            maxSum = Math.max(maxSum, curSum);
        }

        return maxSum;
    }
}

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-3.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        L = 0 
        R = 0 

        if len(nums) == 1: 
            return 1

        while R < len(nums): 

            if nums[R - 1] != nums[R]:
                nums[L] = nums[R]
                L += 1
            
            R += 1
        
        return L